resolver/src/main/java/io/netty/resolver/HostsFileNameResolver.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.internal.EmptyArrays;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link InetNameResolver} that resolves a host name against the hosts file.
+ */
+public class HostsFileNameResolver extends InetNameResolver {
+
+    private static UnknownHostException UNMAPPED_HOST_EXCEPTION =
+            new UnknownHostException("Host isn't mapped in hosts file");
+
+    static {
+        UNMAPPED_HOST_EXCEPTION.setStackTrace(EmptyArrays.EMPTY_STACK_TRACE);
+    }
+
+    private final Map<String, InetAddress> hostsFileEntries;
+
+    /**
+     * @param executor the {@link EventExecutor} which is used to notify the listeners of the {@link Future} returned
+     *                 by {@link #resolve(String)}
+     */
+    public HostsFileNameResolver(EventExecutor executor) {
+        this(executor, HostsFileParser.parseSilently());
```

Comment:
I wonder if we want to schedule a task that re-parse the file every X interval ?

Reply:
> 
> 
> Entirely feasible, but I'd rather save that for another time/someone else.

@slandelle @normanmaurer Could you enlighten if scheduled refresh / failure attempt based refresh is implemented in the current releases?